Wangscape/logging/Logging.cpp

```diff
@@ -0,0 +1,66 @@
+#include "Level.h"
+#include "Logging.h"
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <ostream>
+
+namespace logging
+{
+
+namespace
+{
+
+std::ofstream out;
+Level currentLevel = Level::Info;
+Logger noopLogger;
+Logger currentLogger(&std::cout);
+
+} // anonymous namespace
+
+void setAppender(AppenderType appenderType, std::string name)
+{
+    switch (appenderType)
+    {
+        case AppenderType::StdOut:
+        {
+            currentLogger.setStream(&std::cout);
+            break;
+        }
+        case AppenderType::StdErr:
+        {
+            currentLogger.setStream(&std::cerr);
+            break;
+        }
+        case AppenderType::File:
+        {
+            if (out.is_open()) {
+                out.close();
+            }
+            out.open(name, std::ios_base::app);
```

Comment:
I'd rather want to use it as logger name, which is at the same time name of a file it's writing to. I see now, how bad designed it is :wink:. Overload will work a bit better here, I think.